packages/block-library/src/blocks/sidecar/frontend.js

```diff
@@ -0,0 +1,173 @@
+        contentBlocks = content.children,
+        sidebarBlocks = sidebar.children,
+
+        pulledBlocksArray = Array.from(pulledBlocks),
+        contentBlocksArray = Array.from(contentBlocks),
+        sidebarBlocksArray = Array.from(sidebarBlocks),
+        alignedBlocks = contentBlocksArray.filter ( (block) => ALIGN_CLASSES.some( ALIGN_CLASS => block.classList.contains(ALIGN_CLASS) && ! block.classList.contains( SIDECAR_CLASS ) )),
+        breakingBlocks = alignedBlocks.concat(pulledBlocksArray),
+
+        sidebarIsLeft = content.parentElement.classList.contains( 'novablocks-sidecar--sidebar-left' ),
+
+        // Overlapping between content blocks and sidebar blocks.
+        overlappingBlocks = generateOverlappingBlocks( breakingBlocks, sidebarBlocksArray );
+
+    overlappingBlocks.forEach( block => {
+
+      let noCollisionClass = sidebarIsLeft ? BREAK_LEFT_CLASS : BREAK_RIGHT_CLASS;
+
+      if ( block.classList.contains( PULL_RIGHT_CLASS ) ) {
+        noCollisionClass = BREAK_RIGHT_CLASS;
+      }
+
+      if ( block.classList.contains( PULL_LEFT_CLASS ) ) {
+        noCollisionClass = BREAK_LEFT_CLASS;
+      }
+
+      block.classList.add( noCollisionClass );
+
+    } )
+
+    recalculateOverlappedBlocks( sidecar, overlappingBlocks);
+  } )
+}
+const debouncedSidecarTransformations = debounce(handleSidecarTransformations, 200)
+
+// Helper function to check
+// if two elements are overlapping
+function doesOverlap( elem, collider ) {
+
+  const elemBox = elem.getBoundingClientRect();
+  const colliderBox = collider.getBoundingClientRect();
+
+  const overlap = colliderBox.top + colliderBox.height <= elemBox.top ||
+                  colliderBox.top >= elemBox.top + elemBox.height;
+
+  return ! overlap;
+}
+
+// Helper function to generate overlapping blocks,
+// based on two areas
+// We will create an array with
+// blocks from primaryArea that are overlapping with
+// blocks from secondaryArea.
+
+function generateOverlappingBlocks(primaryArea, secondaryArea) {
+
+  let array = []
+
+  primaryArea.forEach( primaryAreaBlock => {
+
+    secondaryArea.forEach( secondaryAreaBlock => {
+
+      // Avoid useless iterations.
+      if ( array.includes( primaryAreaBlock ) ) {
+        return;
+      }
+
+      if ( doesOverlap(primaryAreaBlock, secondaryAreaBlock)) {
+        array.push(primaryAreaBlock);
+      }
+    })
+  })
+
+  return array;
+}
+
+// We want to listen to Content Width setting
+// change inside Customizer Preview,
+// so we can break wide and full elements
+// if there is not enough available space.
+function sidecarTransformationsInCustomizer() {
+
+  if ( wp.customize !== undefined ) {
+    wp.customize( `${customify.config.options_name}[content_wide_width_addon]`, ( setting ) => {
+      setting.bind( value => {
+        debouncedSidecarTransformations()
+      } );
+    } )
+  }
+}
+
+
+function recalculateOverlappedBlocks( sidecar, blocks ) {
```

Comment:
Can this operation be made inside the `overlappingBlocks.forEach( block => {` loop to avoid adding and then removing the same classes in the same run of the `handleSidecarTransformations` function?